chips/atmega2560-hal/src/adc.rs

```diff
@@ -0,0 +1,56 @@
+extern crate avr_hal_generic as avr_hal;
+
+use crate::port::portf::{PF0, PF1, PF2, PF3, PF4, PF5, PF6, PF7};
+use crate::port::portk::{PK0, PK1, PK2, PK3, PK4, PK5, PK6, PK7};
+
+use crate::atmega2560::adc::admux::MUX_A;
+
+avr_hal::impl_adc! {
+    pub struct Adc {
+        type ChannelID = MUX_A;
+        peripheral: crate::atmega2560::ADC,
+        set_mux: |peripheral, id| {
+            peripheral.admux.modify(|_, w| w.mux().variant(id));
```

Comment:
There's a hidden issue here ... the `ADMUX[MUX]` field is only 5 bits wide; the 6th bit is `ADCSRB[MUX5]`.  However, the enumerated values for for `ADMUX[MUX]` contain _all_ 6bit values valid for the combination of the two.  The problem is than when writing here, the uppermost bit will get cut off and ignored.  Leading to the wrong channels being read for MUX values >31.  E.g. `ADC8` will actually read `ADC0`, `ADC9` will read `ADC10`, etc.

This is obviously a problem and the most correct way to deal with this would be to remove all enumerated values >31 with an SVD patch so you can't accidentally do this in the first place.  This would then mean you'll have to approach this similar to what I did in [`atmega32u4_hal::adc`](https://github.com/Rahix/avr-hal/blob/e3a39bf14f349cc13b916ac6910330cdc38e268a/chips/atmega32u4-hal/src/adc.rs#L7-L29) where I redefined the enumerated values on my own.

However, this is a lot of unnecessary work IMO when we can do a quicker fix here by just extracting the uppermost bit and setting that separately:

```suggestion
            peripheral.admux.modify(|_, w| w.mux().variant(id));
            // Maybe add a comment why this is necessary ...
            peripheral.adcsrb.modify(|_, w| w.mux5().bit((id as u8) & 0b100000 != 0));
```

Reply:
Indeed!  That's a pretty clever way to address the issue.

I confirmed that without this change, attempts to read single-ended ADC8-15 results in reads of ADC0-7, and that with the fix in place the reads are correct.  I made no attempts to test differential reads.

Reply:
Sounds good!